### ISSUE TITLE
Not working with HA Core 2023.5.0+

### ISSUE THREAD

**Issue by AUSF** [richardzone/homeassistant-dht#61]
The integration seems to have stopped working after the last HA Core update. In the log file I only have many of these warning messages:

Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.xxx(s), this was the 1st time calling it.

Thanks for your help!

----------
Home Assistant 2023.5.1
Supervisor 2023.04.1
Operating System 10.1
Frontend 20230503.2 - latest

**Comment by llacoste** [comment]
I am seeing the same behavior.

**Comment by llacoste** [comment]
After some time, despite the warnings, the sensors seem to be functioning well. Not sure if these warnings matter.

**Comment by AUSF** [comment]
> After some time, despite the warnings, the sensors seem to be functioning well. Not sure if these warnings matter.

Interesting...in my case the entities are indeed still available in HA, but they are not updated.

**Comment by llacoste** [comment]
That said, I have had no luck with DHT22 sensors. DHT11 sensors seem to still be working.

**Comment by walberjunior** [comment]
I managed to solve using the
https://github.com/thecode/ha-rpi_gpio/tree/config-flow

I still need to suppress the errors from the logs, but it seems to be working fine.

````
logs:
  custom_components.dht.sensor: error

````

````
2023-05-17 14:49:33.851 WARNING (SyncWorker_6) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.443(s), this was the 1st time calling it.
2023-05-17 14:50:32.145 WARNING (SyncWorker_0) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
2023-05-17 14:50:32.502 WARNING (SyncWorker_6) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.354(s), this was the 1st time calling it.
2023-05-17 14:51:02.143 WARNING (SyncWorker_3) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.353(s), this was the 1st time calling it.
2023-05-17 14:51:02.502 WARNING (SyncWorker_10) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
2023-05-17 14:51:32.137 WARNING (SyncWorker_1) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
2023-05-17 14:51:32.541 WARNING (SyncWorker_2) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
2023-05-17 14:52:02.145 WARNING (SyncWorker_3) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.353(s), this was the 1st time calling it.
2023-05-17 14:52:02.504 WARNING (SyncWorker_9) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.355(s), this was the 1st time calling it.
2023-05-17 14:52:32.148 WARNING (SyncWorker_0) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
2023-05-17 14:52:32.505 WARNING (SyncWorker_8) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
2023-05-17 14:53:02.549 WARNING (SyncWorker_8) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.355(s), this was the 1st time calling it.
2023-05-17 14:53:32.517 WARNING (SyncWorker_2) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
2023-05-17 14:54:02.159 WARNING (SyncWorker_3) [custom_components.dht.sensor] Finished call to 'custom_components.dht.sensor.DHTClient.read_retry' after 0.352(s), this was the 1st time calling it.
````

**Comment by walberjunior** [comment]
Changing the ha-rpi_gpio integration from main to config-flow was the only way that solved my problem after the 2023.5.xx updates.

And according to fer4droid, both integrations are necessary
`````
https://community.home-assistant.io/t/dht-sensor-custom-components/390428/48?u=walberjunior
